File: src/main/java/com/backfield/imagegenerator/DrawLowerLeftTriangle.java
package com.backfield.imagegenerator;

import java.awt.image.BufferedImage;

public class DrawLowerLeftTriangle implements Drawable {
    @Override
    public void draw(BufferedImage image, int color, int x, int y) {
        for (int squareX = 0; squareX < ImageDraw.squareSize; squareX++) {
            for (int squareY = squareX; squareY < ImageDraw.squareSize; squareY++) {
                image.setRGB(x * ImageDraw.squareSize + squareX, y * ImageDraw.squareSize + squareY, color);
            }
        }
    }
}

File: src/main/scala/com/backfield/imagegenerator/DrawLowerLeftTriangle.scala
package com.backfield.imagegenerator

;

import java.awt.image.BufferedImage;

/**
 * Created by jbackfield on 3/12/15.
 */
object DrawLowerLeftTriangle {

  def draw(image: BufferedImage, color: Int, x: Int, y: Int, squareSize: Int): Unit = {
    for (squareX <- 0 until squareSize; squareY <- squareX until squareSize) {
      image.setRGB(
        x * squareSize + squareX,
        y * squareSize + squareY,
        color
      )
    }
  }
}
